File: Dk.Odense.SSP/SSP-UI/src/app/Shared/Services/robustness.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from '@services/base.service';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Robustness } from '@domain-models/robustness.model';

@Injectable({
  providedIn: 'root',
})
export class RobustnessService extends BaseService<Robustness> {
  constructor(http: HttpClient) {
    super(http);
    this.endpoint = 'robustness';
  }

  getNext(personId: string, increment: number): Observable<Robustness> {
    return this.getHTTPService().get<Robustness>(
      this.baseUrl +
        this.endpoint +
        '/getnextbyperson?personId=' +
        personId +
        '&increment=' +
        increment
    );
  }

  getPrevious(personId: string, increment: number): Observable<Robustness> {
    return this.getHTTPService().get<Robustness>(
      this.baseUrl +
        this.endpoint +
        '/getpreviousbyperson?personId=' +
        personId +
        '&increment=' +
        increment
    );
  }
}

File: Dk.Odense.SSP/Dk.Odense.SSP.Application/RobustnessService.cs
using System;
using System.Threading.Tasks;
using Dk.Odense.SSP.Application.Interfaces;
using Dk.Odense.SSP.Domain.Interfaces;
using Dk.Odense.SSP.Domain.Model;

namespace Dk.Odense.SSP.Application
{
    public class RobustnessService : BaseService<Robustness, IRobustnessRepository>, IRobustnessService
    {
        private readonly IRobustnessRepository robustnessRepository;

        public RobustnessService(IRobustnessRepository repository) : base(repository)
        {
            this.robustnessRepository = repository;
        }

        public async Task<int> GetCountByPersonId(Guid personId)
        {
            return await robustnessRepository.GetCountByPersonId(personId);
        }

        public Task<Robustness> GetNext(Guid personId, int increment = 0)
        {
            var res = Repository.GetNext(personId, increment);

            return res;
        }

        public async Task<Robustness> GetPrevious(Guid personId, int increment = 0)
        {
            var res = await Repository.GetPrevious(personId, increment);

            return res;
        }
    }
}
